### ISSUE TITLE
AWS like search bar directive

### ISSUE THREAD

**Issue by manuasir** [wazuh/wazuh-splunk#323]
![](https://user-images.githubusercontent.com/8860227/47663555-991d9000-db9d-11e8-805d-701040118794.gif)

It would be nice to have a similar search bar for our app. This task is divided into the next points:

### Iteration 1 (49th week sprint)

- [x] It must be an Angular.js directive (reusable, parameterizable)
- [x] It must accept a parameter where we define the suggestions that the search bar will show
- [x] When adding filters it should work as follow: if there is a similar filter, the new filter should be an OR operation with the existing ones. If there is no similar filter, the new filter must be an AND operation with the existing ones. Example: `(group: default OR group: staging) AND os: Linux`
- [x] Filters must be removable.
- [x] Remove all the current selectors. Example: the agents' list has some selectors, their logic should be supplied by this new directive.
- [x] Agents list section

https://github.com/wazuh/wazuh-splunk/pull/399/

I may forget some other list from our Wazuh API, but definitively all our lists must use this new one directive. Also, all our related selectors must disappear.

Most of the logic evolved while fetching data should use the queries feature (`q=`) from the Wazuh API.

**Comment by manuasir** [comment]
Implementation for `Agents` list here: https://github.com/wazuh/wazuh-splunk/pull/399

**Comment by manuasir** [comment]
Several endpoints don't currently allow `q` search (blocked by wazuh/wazuh-api#248)
